src/cmds/qsub.c

```diff
@@ -5485,7 +5485,8 @@ do_daemon_stuff(void)
 
 out:
 	close(bindfd);
-	unlink(fl);
+	if (cred_timeout != 1)
```

Comment:
why don't you do what do_daemon_stuff does for windows i.e. just goto out label? I also find it interesting that the place we check that the credentials have expired, we unlink the file and then continue receiving data from the foreground process.

Reply:
@arungrover yes this is part of the out label itself, the issue was that we were unlinking the socket file created by the next daemon that would have come up (so removing the file twice). 
Yes, we unlink the socket file first, so that no new qsub can connect to this daemon any more as it has decided to quit soon, but then continue to read and process data that was "already sent" to this socket by a already connected qsub front-end..

Reply:
Thanks, @subhasisb for the explanation. I am still a little confused about the implementation. The way we are exiting the background process is different between windows and linux. Maybe it is because in windows things work differently but I just want to be sure.

In case of windows, as soon as we see that the credentials are about to expire then we just exit from the process but in case of linux we process any pending request and then exit. To maintain consistency one of them should change, either in Windows, we should not cleanup so quickly, serve pending request and also provide 5 seconds of grace period for foreground process to exit OR in Linux we should just exit as soon as we see credentials are expiring and send DMN_REFUSE_EXIT or something similar to foreground process before exiting.

Reply:
@arungrover Please check https://github.com/PBSPro/pbspro/pull/613 for fix details.
The Bug fix was targeted differently for both the platforms. This part of code affects only Linux. 

Explaination from Praveen on the PR: 

Fix on Windows platform:

1. Problem is consistently reproducible on Windows platform.
2. I tried incorporating the similar fix that of Linux i.e. disconnecting the pipe and retaining it for
   seconds before closing it.
3. But, the behaviour of DisconnectNamedPipe API on windows is little different than that of unlink() call on linux. Unlinking the UD socket on linux still retains the existing connection object, but DisconnectNamedPipe doesn’t do that. It just allows the client to re-connect before doing the closehandle.
4. So, I just moved the code for checking the Credential timeout (30 mins) much before WaitForMultipleObjects call Just after while(1) loop and problem is resolved. I could see a different qsub process id coming up after 29 mins of cred time-out and no error message is being displayed.

Reply:
Thanks @visheshh I now understand why it was done this way. I feel we could have also fixed the problem if the process becoming the daemon would unlink the file itself before opening the socket. By doing that we could have the code same in both places.

Reply:
@arungrover Do you think we can go ahead with the current fix ?

Reply:
It is your choice, you already have signoff on the change. It just felt inconsistent to me but then this change was not introduced by you in the first place.
#613  just does not seems like the right change to me.